src/com/fediroryshchuk/homework7/threads/synchronize/phaser/TestPhaser.java

```diff
@@ -0,0 +1,36 @@
+package com.fediroryshchuk.homework7.threads.synchronize.phaser;
+
+import java.util.concurrent.Phaser;
+
+public class TestPhaser implements Runnable {
+    Phaser phas;
+    String name;
+
+    TestPhaser(Phaser phaser, String name) {
+        this.phas = phaser;
+        this.name = name;
+        phaser.register();
+        new Thread(this).start();
+    }
+
+    @Override
+    public void run() {
+        System.out.println("Thread " + name + " begins first circle");
+        phas.arriveAndAwaitAdvance();
+        try {
+            Thread.sleep(100);
```

Comment:
You could move this code into a separate util method and reuse anywhere you need it.

Reply:
You didn't resolve the previous comment.  You should change code or answer why not.